### PR TITLE
LPS-126912 Migrate Commerce account roles to Portal account roles

### PR DESCRIPTION
**Pull request by pei-jung**

### CODE REVIEW

**Review by drewbrokke — APPROVED** [review]
Looks good @pei-jung. Were you able to manually test this UpgradeProcess?

@riccardo-alberti please have a look at this and see if it meets your needs. If possible, see if you can test it with a populated database.

**Comment by drewbrokke** [comment]
ci:test:upgrade

**Comment by drewbrokke** [comment]
@achaparro will you please have a look at this change? It's a fairly complex upgrade process so we would appreciate your feedback on it :)

**Comment by achaparro** [comment]
@drewbrokke I will check it on Monday, I am on PTO today :-)

**Comment by riccardo-alberti** [comment]
@drewbrokke checking now! thanks

**Comment by riccardo-alberti** [comment]
@drewbrokke just tested after creating Minium Demo on 7.3 and upgrading... i got this error on postgres

`Caused by: com.liferay.portal.kernel.log.LogSanitizerException: org.postgresql.util.PSQLException: ERROR: improper qualified name (too many dotted names): com.liferay.account>
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440) ~[postgresql.jar:42.2.5]
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183) ~[postgresql.jar:42.2.5]
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308) ~[postgresql.jar:42.2.5]
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441) ~[postgresql.jar:42.2.5]
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365) ~[postgresql.jar:42.2.5]
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:143) ~[postgresql.jar:42.2.5]
        at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:106) ~[postgresql.jar:42.2.5]
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52) ~[hikaricp.jar:?]
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java) ~[hikaricp.jar:?]
        at com.liferay.commerce.account.internal.upgrade.v9_1_0.CommerceAccountRoleUpgradeProcess.doUpgrade(CommerceAccountRoleUpgradeProcess.java:86) ~[?:?]
        at com.liferay.portal.kernel.upgrade.UpgradeProcess.lambda$upgrade$0(UpgradeProcess.java:114) ~[portal-kernel.jar:?]
        at com.liferay.portal.db.partition.DBPartitionUtil.forEachCompanyId(DBPartitionUtil.java:113) ~[portal-impl.jar:?]
        at com.liferay.portal.dao.db.BaseDB.process(BaseDB.java:307) ~[portal-impl.jar:?]
        at com.liferay.portal.kernel.dao.db.BaseDBProcess.process(BaseDBProcess.java:225) ~[portal-kernel.jar:?]
        at com.liferay.portal.kernel.upgrade.UpgradeProcess.upgrade(UpgradeProcess.java:101) ~[portal-kernel.jar:?]
        ... 146 more`

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
@riccardo-alberti I updated the query to fix the error you posted. Can you test again? Thanks!

**Comment by drewbrokke** [comment]
@achaparro did you have a chance to look at this one?

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						"update Role_ set type_ = " +
+							RoleConstants.TYPE_ACCOUNT + " where roleId = ?"));
+			PreparedStatement preparedStatement3 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						StringBundler.concat(
+							"update UserGroupRole inner join Group_ on ",
+							"UserGroupRole.groupId = Group_.groupId and ",
```
Comment (achaparro) [inline modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:81]: Like in the previous query, we should filter first by classNameId

File: modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						"update Role_ set type_ = " +
+							RoleConstants.TYPE_ACCOUNT + " where roleId = ?"));
+			PreparedStatement preparedStatement3 =
+				AutoBatchPreparedStatementUtil.autoBatch(
+					connection.prepareStatement(
+						StringBundler.concat(
+							"update UserGroupRole inner join Group_ on ",
```
Comment (achaparro) [inline modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v9_1_0/CommerceAccountRoleUpgradeProcess.java:80]: As far as I remember UPDATES with inner join have different syntax in Mysql and MariaDB than in the other databases.
I believe the syntax you have used will only work on MySQL and MariaDB. Please check these links:
https://stackoverflow.com/questions/8331687/mysql-update-a-joined-table
https://stackoverflow.com/questions/44118923/update-with-inner-join-postgresql

And we already addressed this issue in here:
https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/upgrade/v7_0_0/UpgradeUser.java

**Review by achaparro — CHANGES_REQUESTED** [review]
Hi @drewbrokke, @pei-jung, @riccardo-alberti,

I verified the upgrade changes and I made some comments, the most important one is the verification of the update with an inner join, please check it.

Thanks.

**Comment by pei-jung** [comment]
@achaparro thank you for taking the time to review the changes. I've applied the changes you suggested. Please see https://github.com/liferay-usm/liferay-portal/pull/271. Thanks!

**Pull request closed without merging**